pom.xml

```diff
@@ -125,7 +125,7 @@ under the License.
 		<junit4.version>4.13.2</junit4.version>
 		<junit5.version>5.8.1</junit5.version>
 		<archunit.version>0.22.0</archunit.version>
-		<mockito.version>2.21.0</mockito.version>
+		<mockito.version>3.4.6</mockito.version>
```

Comment:
Version upgrades should have the separate issue.

cc @zentol

Reply:
This is just a POC and open for discussion. Please see [this](https://github.com/apache/flink/pull/19372#discussion_r849691193) comment where I've considered all of the possible solutions which I'm aware of. As a result static method mocking seems like the least painful solution if we want to mock `UGI`.

All in all if we decide not to do that I can remove the extra commit which contains the Mockito version upgrade. As a consequence either we drop `testStartTGTRenewalShouldScheduleRenewal` or I would like to hear a different approach to mock `UGI`. If we decide to go on w/ this then I agree this must be a separate jira.

I'm basically fine w/ either approaches though my personal opinion is that in this exceptional case static function mocking would be the least painful solution with super compact code.

Reply:
I guess for this case it's justifiable as the Hadoop doesn't provide a reasonable interface you could interact with wrt testing.